client-ui/src/app/components/dashboard/dashboard.url-state-persister.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Params} from "@angular/router";
+import * as DashboardActions from "../../store/actions/dashboard.actions";
+import {DashboardVM} from "../../model/viewModels/dashboardVM";
+import * as RouterAction from "../../store/actions/router.actions";
+import {AppState} from "../../model/app-state";
+import {Store} from "@ngrx/store";
+
+export class DashboardUrlStatePersister {
+  constructor(private store: Store<AppState>) {
+  }
+
+  public persistState(dashboardState: DashboardVM) {
+    console.log("PERSIST STATE TO URL", dashboardState)
+    this.store.dispatch(new RouterAction.ReplaceUrlState({
+      q: JSON.stringify(dashboardState)
+    }))
+  }
+
+  public restoreState(params: Params) {
+    console.log("RESTORE STATE FROM URL", JSON.parse(params.q))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Calls to 'console.log' are not allowed.](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)